### PR TITLE
GameOfLife: Replace DeprecatedString with String

### PR DESCRIPTION
**Pull request by tcl3**

### CODE REVIEW

File: Userland/Games/GameOfLife/Pattern.cpp [hunk 1]
```diff
@@ -24,13 +23,13 @@ void Pattern::set_action(GUI::Action* action)
...
     for (size_t i = 0; i < m_pattern.first().length(); i++) {
         StringBuilder builder;
         for (int j = m_pattern.size() - 1; j >= 0; j--) {
-            builder.append(m_pattern.at(j).substring(i, 1));
+            builder.append(m_pattern.at(j).substring_view(i, 1));
         }
-        rotated.append(builder.to_deprecated_string());
+        rotated.append(builder.string_view());
     }
     m_pattern = move(rotated);
```
Comment (Hendiadyoin1) [inline Userland/Games/GameOfLife/Pattern.cpp:34]: This smells like it'll cause a UAF

Comment (tcl3) [reply Userland/Games/GameOfLife/Pattern.cpp]: You're absolutely right :facepalm:. I've modified the PR to construct `Pattern` objects from `String` instead and made sure that rotating actually works this time!

File: Userland/Games/GameOfLife/Pattern.cpp [hunk 2]
```diff
@@ -24,13 +23,13 @@ void Pattern::set_action(GUI::Action* action)
...
-    for (size_t i = 0; i < m_pattern.first().length(); i++) {
+    Vector<String> rotated;
+    for (size_t i = 0; i < m_pattern.first().bytes_as_string_view().length(); i++) {
         StringBuilder builder;
         for (int j = m_pattern.size() - 1; j >= 0; j--) {
-            builder.append(m_pattern.at(j).substring(i, 1));
+            builder.append(m_pattern.at(j).bytes_as_string_view().substring_view(i, 1));
         }
-        rotated.append(builder.to_deprecated_string());
+        rotated.append(builder.to_string().release_value_but_fixme_should_propagate_errors());
```
Comment (tcl3) [inline Userland/Games/GameOfLife/Pattern.cpp:32]: Given that we know this is a small allocation and bearing #20405 in mind, should `.release_value_but_fixme_should_propagate_errors()` the best choice here or would `.release_value()` or `MUST` be better?

Comment (ADKaster) [reply Userland/Games/GameOfLife/Pattern.cpp:32]: I would go for MUST

**Review by ADKaster — APPROVED** [review]

**Pull request merged**